### PR TITLE
test get repos + fix missing event check on get repos api in github and gitlab

### PR DESCRIPTION
**Pull request by olgakil**

### CODE REVIEW

**Comment by cxflowtestuser** [comment]
Scan submitted to Checkmarx

**Comment by cxflowtestuser** [comment]
Logo<br />**Checkmarx SAST - Scan Summary & Details**
### Cx-SAST Summary
Total of 92 vulnerabilities<br />High&nbsp;**0 High**<br />Medium&nbsp;**10 Medium**<br />Low&nbsp;**75 Low**<br />Info&nbsp;**7 Info**<br />
### Violation Summary
Medium&nbsp;**6** **Medium**<br />
[View more details on Checkmarx UI](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1049085&projectid=734)
### Cx-SAST Details
|Lines|Severity|Category|File|Link|
---|---|---|---|---
[77](https://github.com/checkmarx-ltd/Integrations-ReposManager/blob/pr-orly-test-getrepos/src/main/java/com/checkmarx/controller/WebController.java#L77) |Medium|Unchecked_Input_for_Loop_Condition|src/main/java/com/checkmarx/controller/WebController.java|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1049085&projectid=734&pathid=9)
[90](https://github.com/checkmarx-ltd/Integrations-ReposManager/blob/pr-orly-test-getrepos/src/main/java/com/checkmarx/service/AbstractScmService.java#L90) |Medium|Privacy_Violation|src/main/java/com/checkmarx/service/AbstractScmService.java|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1049085&projectid=734&pathid=4)
[84](https://github.com/checkmarx-ltd/Integrations-ReposManager/blob/pr-orly-test-getrepos/src/main/java/com/checkmarx/utils/Converter.java#L84) [102](https://github.com/checkmarx-ltd/Integrations-ReposManager/blob/pr-orly-test-getrepos/src/main/java/com/checkmarx/utils/Converter.java#L102) |Medium|Privacy_Violation|src/main/java/com/checkmarx/utils/Converter.java|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1049085&projectid=734&pathid=3)
[136](https://github.com/checkmarx-ltd/Integrations-ReposManager/blob/pr-orly-test-getrepos/src/main/java/com/checkmarx/controller/WebController.java#L136) [169](https://github.com/checkmarx-ltd/Integrations-ReposManager/blob/pr-orly-test-getrepos/src/main/java/com/checkmarx/controller/WebController.java#L169) |Medium|Excessive_Data_Exposure|src/main/java/com/checkmarx/controller/WebController.java|[Checkmarx](https://sast.checkmarx.net/CxWebClient/ViewerMain.aspx?scanid=1049085&projectid=734&pathid=1)
***
Logo
**Checkmarx SCA - Scan Summary & Details**
### Cx-SCA Summary
**Total Packages Identified**: **97**
**Scan Risk Score**: **9.80**

High&nbsp;**2** **High** **severity vulnerabilities**
Medium&nbsp;**1** **Medium** **severity vulnerabilities**
Low&nbsp;**0** **Low** **severity vulnerabilities**
[View more details on Checkmarx UI](https://eu.sca.checkmarx.net/#/projects/3a1896df-7a7c-46d2-b1a1-de303f21ea89/reports/302fb94e-1f1c-40f9-a558-b9a9af5c74b9)
### Cx-SCA vulnerability result overview
|Vulnerability ID|Package|Severity|CVSS score|Publish date|Current version|Recommended version|Link in CxSCA|Reference – NVD link|
---|---|---|---|---|---|---|---|---
`CVE-2016-1000027`|org.springframework:spring-web|HIGH|9.8|2020-01-02T23:15:00|5.2.9.RELEASE|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/3a1896df-7a7c-46d2-b1a1-de303f21ea89/reports/302fb94e-1f1c-40f9-a558-b9a9af5c74b9/vulnerabilities/CVE-2016-1000027%3AYarn-org.springframework%3Aspring-web-5.2.9.RELEASE/vulnerabilityDetails)|[CVE-2016-1000027](https://nvd.nist.gov/vuln/detail/CVE-2016-1000027)
`CVE-2016-1000027`|org.springframework:spring-webmvc|HIGH|9.8|2020-01-02T23:15:00|5.2.9.RELEASE|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/3a1896df-7a7c-46d2-b1a1-de303f21ea89/reports/302fb94e-1f1c-40f9-a558-b9a9af5c74b9/vulnerabilities/CVE-2016-1000027%3AYarn-org.springframework%3Aspring-webmvc-5.2.9.RELEASE/vulnerabilityDetails)|[CVE-2016-1000027](https://nvd.nist.gov/vuln/detail/CVE-2016-1000027)
`CVE-2020-15250`|junit:junit|MEDIUM|4.4|2020-10-11T23:00:00|4.13|| [Vulnerability Link](https://eu.sca.checkmarx.net/#/projects/3a1896df-7a7c-46d2-b1a1-de303f21ea89/reports/302fb94e-1f1c-40f9-a558-b9a9af5c74b9/vulnerabilities/CVE-2020-15250%3AYarn-junit%3Ajunit-4.13/vulnerabilityDetails)|[CVE-2020-15250](https://nvd.nist.gov/vuln/detail/CVE-2020-15250)

File: src/test/java/com/checkmarx/cxintegrations/reposmanager/api/getrepos/getReposApiRunner.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
+package com.checkmarx.cxintegrations.reposmanager.api.getrepos;
+
+import io.cucumber.junit.Cucumber;
+import io.cucumber.junit.CucumberOptions;
+import org.junit.runner.RunWith;
+
+@RunWith(Cucumber.class)
+@CucumberOptions(
+        features = "src/test/resources/cucumber/get-repos-api.feature",
+        glue = {"com.checkmarx.cxintegrations.reposmanager.api.getrepos" } )
```
Comment (alex-ko-dev) [inline src/test/java/com/checkmarx/cxintegrations/reposmanager/api/getrepos/getReposApiRunner.java:10]:
```suggestion
        glue = {"com.checkmarx.cxintegrations.reposmanager.api.getrepos" } )
```
This line is not required.

File: src/test/java/com/checkmarx/cxintegrations/reposmanager/api/getrepos/getReposApiRunner.java [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+public class getReposApiRunner {
```
Comment (alex-ko-dev) [inline src/test/java/com/checkmarx/cxintegrations/reposmanager/api/getrepos/getReposApiRunner.java:11]: Does `mvn test` run this feature file? I remember that class name must have the `Test` postfix.

File: src/test/resources/cucumber/get-repos-api.feature [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+    As a ReposManager API client
+    I want to get all repositories for the given organization and their webhooks
+
+  Scenario Outline: Get Repositories API
+    When API get repositories is called with scm "<scm>"
+    And number of returned repositories will be <n_repos>
+    And There are <n_hooks> webhooks defined on the repositories
+    And number of active hooks is "<n_active_hooks>"
+    And number of hooks with CxFlow url is <n_hooks_with_url>
+    And number of hooks with valid events will be <n_hooks_with_events>
```
Comment (alex-ko-dev) [inline src/test/resources/cucumber/get-repos-api.feature:12]: Here we should explain what 'valid events' are.

File: src/test/resources/cucumber/get-repos-api.feature [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+    Then repositories list returned by CxIntegration will be <n_repos>
+    And number of effective hooks will be <n_effective_hooks>
```
Comment (alex-ko-dev) [inline src/test/resources/cucumber/get-repos-api.feature:14]:
```suggestion
    And the number of repos with 'is webhook enabled' property set to true will be <n_enabled_repos>
```

**Review by alex-ko-dev — APPROVED** [review]

**Pull request merged**